src/analysis/mellin_transform.lean

```diff
@@ -16,29 +17,169 @@ differentiable in a suitable vertical strip.
+end
+
+lemma mellin_const_smul (f : ℝ → E) (s : ℂ)
+  {𝕜 : Type*} [nontrivially_normed_field 𝕜] [normed_space 𝕜 E] [smul_comm_class ℂ 𝕜 E] (c : 𝕜) :
+  mellin (λ t, c • f t) s = c • mellin f s :=
+by simp only [mellin, smul_comm, integral_smul]
+
+lemma mellin_div_const (f : ℝ → ℂ) (s a : ℂ) :
+  mellin (λ t, f t / a) s = mellin f s / a :=
+by simp_rw [mellin, smul_eq_mul, ←mul_div_assoc, integral_div]
+
+lemma mellin_comp_rpow (f : ℝ → E) (s : ℂ) {a : ℝ} (ha : a ≠ 0) :
+  mellin (λ t, f (t ^ a)) s = |a|⁻¹ • mellin f (s / a) :=
+begin
+  -- note: this is also true for a = 0 (both sides are zero), but this is mathematically
+  -- uninteresting and rather time-consuming to check
+  simp_rw mellin,
+  conv_rhs { rw [←integral_comp_rpow_Ioi _ ha, ←integral_smul] },
+  refine set_integral_congr measurable_set_Ioi (λ t ht, _),
+  dsimp only,
+  rw [←mul_smul, ←mul_assoc, inv_mul_cancel, one_mul, ←smul_assoc, real_smul],
+  show |a| ≠ 0, { contrapose! ha, exact abs_eq_zero.mp ha },
+  rw [of_real_cpow (le_of_lt ht), ←cpow_mul_of_real_nonneg (le_of_lt ht),
+    ←cpow_add _ _ (of_real_ne_zero.mpr $ ne_of_gt ht), of_real_sub, of_real_one, mul_sub,
+    mul_div_cancel' _ (of_real_ne_zero.mpr ha), add_comm, ←add_sub_assoc, mul_one, sub_add_cancel]
+end
+
+lemma mellin_comp_mul_left (f : ℝ → E) (s : ℂ) {a : ℝ} (ha : 0 < a) :
+  mellin (λ t, f (a * t)) s = (a : ℂ) ^ (-s) • mellin f s :=
+begin
+  simp_rw mellin,
+  have : eq_on (λ t : ℝ, (t : ℂ) ^ (s - 1) • f (a * t))
+    (λ t : ℝ, (a : ℂ) ^ (1 - s) • (λ u : ℝ, (u : ℂ) ^ (s - 1) • f u) (a * t)) (Ioi 0),
+  { intros t ht,
+    dsimp only,
+    rw [of_real_mul, mul_cpow_of_real_nonneg ha.le (le_of_lt ht), ←mul_smul,
+      (by ring : 1 - s = -(s - 1)), cpow_neg, inv_mul_cancel_left₀],
+    rw [ne.def, cpow_eq_zero_iff, of_real_eq_zero, not_and_distrib],
+    exact or.inl ha.ne' },
+  rw [set_integral_congr measurable_set_Ioi this, integral_smul,
+    integral_comp_mul_left_Ioi _ _ ha, mul_zero, ←complex.coe_smul, ←mul_smul, sub_eq_add_neg,
+    cpow_add _ _ (of_real_ne_zero.mpr ha.ne'), cpow_one, abs_of_pos (inv_pos.mpr ha), of_real_inv,
+    mul_assoc, mul_comm, inv_mul_cancel_right₀ (of_real_ne_zero.mpr ha.ne')]
+end
+
+lemma mellin_comp_mul_right (f : ℝ → E) (s : ℂ) {a : ℝ} (ha : 0 < a) :
+  mellin (λ t, f (t * a)) s = (a : ℂ) ^ (-s) • mellin f s :=
+by simpa only [mul_comm] using mellin_comp_mul_left f s ha
+
+lemma mellin_comp_inv (f : ℝ → E) (s : ℂ) : mellin (λ t, f (t⁻¹)) s = mellin f (-s) :=
+by simp_rw [←rpow_neg_one, mellin_comp_rpow _ _ (neg_ne_zero.mpr one_ne_zero), abs_neg, abs_one,
+  inv_one, one_smul, of_real_neg, of_real_one, div_neg, div_one]
+
+/-- Predicate standing for "the Mellin transform of `f` is defined at `s` and equal to `m`". This
+shortens some arguments. -/
+def has_mellin (f : ℝ → E) (s : ℂ) (m : E) : Prop := mellin_convergent f s ∧ mellin f s = m
+
+lemma mellin_add {f g : ℝ → E} {s : ℂ} (hf : mellin_convergent f s) (hg : mellin_convergent g s) :
```

Comment:
should this be `has_mellin_add`? (And same thing for the next lemma).

Reply:
I would call it `has_mellin.add` to allow dot notation.

Reply:
None of the arguments to `mellin_add` has type `has_mellin` – it is the output, not the input, which has that type – so dot-notation wouldn't work with the current setup.